### PR TITLE
added socket support for joining/leaving a room

### PR DESCRIPTION
**Pull request by rushitote**

### CODE REVIEW

File: server/src/socket/index.ts [hunk 1]
```diff
@@ -69,5 +69,29 @@ export class SocketIO {
         }
       })
     })
+
+    socket.on('userJoinRoom', body => {
+      const { roomId } = JSON.parse(body)
+      const username = (socket.request as socketRequest).user.dataValues.username
+      if (roomId) {
+        joinRoom(username, roomId).then(status => {
```
[the remark this answers is not in the document]
Comment (rushitote) [reply server/src/socket/index.ts]: I have changed this.

**Comment by rushitote** [comment]
So, now the user joining or leaving the room and emitting that real time using socket has been implemented in both front-end and server. I also added a Toast notification feature when some user joins or leaves the room and it looks really great to me. Let me know what you think. 

This is the look:
image
image

File: front-end/src/Pages/JoinRoom.tsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ export default function JoinRoom(props: IJoinRoomProps) {
         } else {
           try {
             addToRoom(roomId)
+            history.push(`/redirecting`)
```
Comment (TheTrio) [inline front-end/src/Pages/JoinRoom.tsx:36]: What's this for?

**Review by TheTrio — APPROVED** [review]
Looks good!

File: front-end/src/Pages/JoinRoom.tsx (hunk 2 above)
Comment (rushitote) [reply front-end/src/Pages/JoinRoom.tsx:36]: Yeah, directly pushing the room to the history didn't seem to fully re-render the page causing the older user list to be displayed, so this redirect solved it. Maybe we can add a loading screen there.

**Pull request merged**